assignments/assignment_2/README.md

```diff
@@ -59,14 +59,14 @@ We know that the things presented in the previous modules are orientative guidel
 This exercise is very simple, we just want you to fish out some current or older project that you are or were part of and review its organization. What would you change? What would you keep? Why? 
 
 
-## 2- Structure recognition
+## 2- Structure analysis
 
-In collecting a couple of examples for the project structure module we made a mess and lost track of the structure each of the projects. Give it a go and see if you can figure out which patterns do these projects follow.
+Again, a simple exercise. The idea this time is to practise your ability to navigate projects you have not written yourself. 
+- How would you go about understanding the different components of a project for the first time? 
+- What if you are looking for a specific feature, class, function, _etc._? How would you proceed then?
 
-[Link to mysterious project 1](https://github.com/qiuyu96/CoDeF/tree/main)
+Enough chitchat, let us get our hands dirty! Your mission is to understand the structure of two of the most popular python libraries:
+- [requests](https://github.com/psf/requests/tree/main),
+- [scikit-learn](https://github.com/scikit-learn/scikit-learn/tree/main). 
```

Comment:
**Suggestion**: I would keep your 3rd project here as well: https://github.com/qiuyu96/CoDeF/tree/main

This example is interesting because it isn't a package, just a bunch of python scripts. Note how there are many entrypoints (one script for train, for hyperparameter optimization, etc) and one will have to read the documentation before being able to use it properly. 

Also note that this ties back to the `pyproject.toml` module: this repo was organized in a way that that all imports come from its sibling modules, so no functionality can be shared from parent modules to their children, only from the children to their parents. This was the first way we listed on how to solve the ModuleNotFound errors, and we don't recommend it: too much work for too little benefit.

Many repos coming out of academia are like this.